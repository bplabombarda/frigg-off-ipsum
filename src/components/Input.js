import React from 'react';
import {
  func,
  string,
} from 'prop-types'

import '../styles/Input.scss';

const Input = ({ onClick, value }) => {
	return (
		<input
      className='button'
      value={ value }
      type='submit'
      onClick={ onClick }
      />
	);
};

Input.propTypes = {
  onClick: func,
  value: string,
};

export default Input;
