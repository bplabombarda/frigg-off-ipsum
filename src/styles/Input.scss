.button {
	background: #333;
	border: none;
	padding: 0.6em;
	color: #ffffff;
	border-radius: 3px;
	font-family: 'Anonymous Pro';
	font-size: 1em;
	font-weight: 700;
	cursor: pointer;
	margin-right: 0.5em;
}

.button:hover {
	background: #474747;
}

.button-disabled,
.button-disabled:hover {
	background-color: #777;
}

.copy-button {
	position: relative;
	display: block;
	color: #d21e25;
	background: #333;
	margin: 1em;
	padding: 0.6em;
	width: 100px;
	margin: 1em auto 0;
	border-radius: 3px;
	cursor: pointer;
}

.copy-button:hover {
	background: #474747;
}
