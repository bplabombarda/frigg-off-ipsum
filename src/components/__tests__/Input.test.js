import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Input from '../Input';

const mockProps = {
  onClick: jest.fn(),
  value: '3',
};

describe('rendering', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Input { ...mockProps }/>
    );
  
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});