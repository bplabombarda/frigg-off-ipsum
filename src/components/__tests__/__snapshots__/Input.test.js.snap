// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering matches snapshot 1`] = `
<input
  className="button"
  onClick={[MockFunction]}
  type="submit"
  value="3"
/>
`;
